CarboPlanner/src/utils/Database/DatabaseActions.tsx

```diff
@@ -9,57 +8,59 @@ let today = new Date();
   let date = today.getDate() + "-"+ parseInt(today.getMonth()+1) +"-"+today.getFullYear()
   const mainDocRef = doc(db, process.env.EXPO_PUBLIC_UUID, date);
 
-  const data = [ ];
+  const data = {};
 
-  const createData = async (newData) => {
+  const saveData = async (uuidKey, newData) => {
     if (!(await getDoc(mainDocRef)).exists()) {
       await setDoc(mainDocRef, {
         data
       });
     };
+    //get exsisting data
+    const docSnapshot = await getDoc(mainDocRef);
+    const exsistingData = docSnapshot.data().data;
+
+    //Add new data into the object 
+    const updateData = {
+      ...exsistingData, [uuidKey]: newData
+    }
 
+    //uopdate document with updated dat
     await updateDoc(mainDocRef, {
-      data: arrayUnion(newData)
+      data: updateData
     });
   };
 
- const edit = async (data, uuid) => {
-    if (!data || !uuid) {
-        console.log("Data and UUID are required.");
-        return;
-    }
+  const edit = async (data, uuid) => {
 
     const docSnapshot = await getDoc(mainDocRef);
     if (docSnapshot.exists()) {
         const fetchedData = docSnapshot.data().data || [];
 
-        let index = -1; // Initialize index as -1
-
-        fetchedData.forEach((obj, i) => {
-            const key = Object.keys(obj)[0]; // Get the UUID key of the object
-
-            if (key === uuid) {
-                index = i; // Update the index if UUID matches
-            }
-        });
-
-        if (index !== -1) {
-            // Update the data at the found index with the provided data
-            fetchedData[index][uuid] = data[uuid];
-            console.log("Updated data:", fetchedData[index]);
-
-            // Update the document with the changes
-            await updateDoc(mainDocRef, { data: fetchedData });
-            console.log("Document updated successfully.");
-        } else {
-            console.log("UUID key not found in data.");
-        }
-    } else {
-        console.log("Document does not exist.");
+        // Update the data at the specific UUID
+        fetchedData[uuid] = data;
+
+        // Update the document with the changes
+        await updateDoc(mainDocRef, { data: fetchedData });
     }
 };
 
-  export {createData, mainDocRef, edit};
+
+const deleteMeal = async (uuidKey) => {
+  const docSnapshot = await getDoc(mainDocRef);
+  const exsistingData = docSnapshot.data().data;
+
+  const updateData = {...exsistingData}
+  delete updateData[uuidKey]
```

Comment:
The naming is a bit confusing
```suggestion
  delete meal[uuidKey]
```